src/random.h

```diff
@@ -67,7 +67,17 @@
  * Thread-safe.
  */
 void GetRandBytes(unsigned char* buf, int num) noexcept;
+/** Generate a random integer in the range [0..range). Precondition: range > 0 */
 uint64_t GetRand(uint64_t nMax) noexcept;
+// The following GetRandDur<duration> MUST NOT be templated on the type of the
+// <duration> to prevent unexpected results. If there is a mismatch between the
+// return type and the argument type, the compiler is unable to reliably select
```

Comment:
Sorry to nitpick here, but i think this is still confusing.

In a hypothetical

```c++
template<typename T>
T GetRandDur(T max) { return T{GetRand(max.count())}; }
```

When `std::chrono::seconds r = GetRandDur(std::chrono::hours{1})` there is no unreliable template selection or mismatch between types. `GetRandDur` is reliably instantiated with `T`=`std::chrono::hours`, and is returning a random std::chrono::hours (the same type). It's in the assignment outside of the function call that the conversion to `std::chrono::seconds` happens, and this assignment doesn't lose precision; the precision was never requested in the first place.

How about this instead:

    Templates cannot be used here to avoid stating explicitly what precision
    is desired in the result. A hypothetical
    `template<typename T> T GetRandDur(T)` invoked on a
    `std::chrono::hours` argument will only ever return a duration with the
    resolution of 1 hours, which may be unexpected when assigning to
    a variable of a higher resolution duration like `std::chrono::seconds`.

Reply:
Thanks. Copy-pasted your text.

Reply:
ah yes. ok this confirms what I was trying to get at in the other thread. thanks!